python/kserve/kserve/protocol/rest/server.py

```diff
@@ -86,67 +101,151 @@ def create_application(self) -> FastAPI:
+                    v2_endpoints.model_metadata,
+                    tags=["V2"],
+                    include_in_schema=False,
+                ),
+                FastAPIRoute(
+                    r"/v2/models/{model_name}/ready",
+                    v2_endpoints.model_ready,
+                    response_model=ModelReadyResponse,
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"v2/models/{model_name}/versions/{model_version}/ready",
+                    v2_endpoints.model_ready,
+                    response_model=ModelReadyResponse,
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"/v2/models/{model_name}/infer",
+                    v2_endpoints.infer,
+                    methods=["POST"],
+                    response_model=InferenceResponse,
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"/v2/models/{model_name}/generate",
+                    v2_endpoints.generate,
+                    methods=["POST"],
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"/v2/models/{model_name}/generate_stream",
+                    v2_endpoints.generate_stream,
+                    methods=["POST"],
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"/v2/models/{model_name}/versions/{model_version}/infer",
+                    v2_endpoints.infer,
+                    methods=["POST"],
+                    tags=["V2"],
+                    include_in_schema=False,
+                ),
+                FastAPIRoute(
+                    r"/v2/repository/models/{model_name}/load",
+                    v2_endpoints.load,
+                    methods=["POST"],
+                    tags=["V2"],
+                ),
+                FastAPIRoute(
+                    r"/v2/repository/models/{model_name}/unload",
+                    v2_endpoints.unload,
+                    methods=["POST"],
+                    tags=["V2"],
+                ),
+            ],
+            exception_handlers={
                 InvalidInput: invalid_input_handler,
                 InferenceError: inference_error_handler,
                 ModelNotFound: model_not_found_handler,
                 ModelNotReady: model_not_ready_handler,
                 NotImplementedError: not_implemented_error_handler,
-                Exception: generic_exception_handler
-            }
+                Exception: generic_exception_handler,
+            },
         )
+        # Register OpenAI endpoints if any of the models in the registry implement the OpenAI inferface
+        # This adds /openai/v1/completions and /openai/v1/chat/completions routes to the
+        # REST server.
+        maybe_register_openai_endpoints(app, self.dataplane.model_registry)
```

Comment:
```python
if _openai:
   maybe_register_openai_endpoints(app, self.dataplane.model_registry)
```

Reply:
This function is safe to call even if `openai` is not available.